backbone/src/main/java/org/researchstack/backbone/ui/BaseActivity.kt

```diff
@@ -0,0 +1,31 @@
+package org.researchstack.backbone.ui
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import org.researchstack.backbone.utils.LocalBroadcaster
+
+open class BaseActivity : AppCompatActivity() {
```

Comment:
Sorry to be the Devil's advocate here, but I don't like using a BaseActivity/Inheritance for this.

We have Jetpack and there's Lifecycle, so I'd rather have a lifecycle aware component (that can be injected or constructed by hand and set up) that receives the lifecycle events and does what it needs to do. 

Also does this only apply to PinCodeActivity? How are TaskActivity, ConsentTaskActivity, etc. going to do this? Are they going to extend a baseActivity in backbone?

Reply:
ConsentTaskActivity and  TaskActivity  extends PinCodeActivity eventually so It going to apply into it.

Reply:
More reason not to add **another** base activity.